src/viewer.js

```diff
@@ -109,34 +114,52 @@
             rMatrix.e = oldE + (targetE - oldE) * step; // TransX
             rMatrix.f = oldF + (targetF - oldF) * step; // TransY
 
-            rects.forEach(function (r) {
-                var rect = r.node;
-                rect.setAttribute('rx', Math.max(0.0, ROUNDRECT_R / scaleX));
-                rect.setAttribute('ry', Math.max(0.0, ROUNDRECT_R / scaleY));
-            });
+            if (rects) {
+                rects.forEach(function (r) {
+                    var rect = r.node;
+                    rect.setAttribute('rx', Math.max(0.0, fig.roundradius / scaleX));
+                    rect.setAttribute('ry', Math.max(0.0, fig.roundradius / scaleY));
+                });
+            }
         };
 
         var finish = function () {
             scaleViewport(1);
             var scaleXt = 1.0 / targetScaleX;
             var scaleYt = 1.0 / targetScaleY;
-            rects.forEach(function (r) {
-                var rect = r.node;
-                var rectx = rect.x.baseVal.value;
-                var recty = rect.y.baseVal.value;
-                var rectw = rect.width.baseVal.value;
-                var text = rect.nextElementSibling;
-                var shortinfo = rect.getAttribute("data-shortinfo");
-
+            var updateText = function (text, x, y, w, shortinfo) {
                 var tMatrix = text.transform.baseVal.getItem(0).matrix;
                 tMatrix.a = scaleXt;
                 tMatrix.d = scaleYt;
-                tMatrix.e = (1.0 - scaleXt) * rectx;
-                tMatrix.f = (1.0 - scaleYt) * recty;
+                tMatrix.e = (1.0 - scaleXt) * x;
+                tMatrix.f = (1.0 - scaleYt) * y;
 
-                text.textContent = formatText(fig, shortinfo, rectw / scaleXt);
+                text.firstChild.nodeValue = formatText(fig, shortinfo, w / scaleXt);
```

Comment:
The `firstChild.nodeValue` is not as safe as `textContent`. We should be careful that this doesn't cause any problems.